pom.xml

```diff
@@ -53,9 +53,14 @@
         <version.org.sonatype.plugins.nexus-staging-maven-plugin>1.6.6</version.org.sonatype.plugins.nexus-staging-maven-plugin>
         <version.checkstyle>8.5</version.checkstyle>
         <version.org.wildfly.checkstyle-config>1.0.7.Final</version.org.wildfly.checkstyle-config>
+        <version.org.jacoco>0.7.8</version.org.jacoco>
 
         <!-- Checkstyle configuration -->
         <linkXRef>false</linkXRef>
+
+        <!-- jvm args -->
+        <jvm.args.jacoco>-javaagent:${project.build.directory}/lib/jacocoagent.jar=destfile=${project.build.directory}/coverage-reports/jacoco-ut.exec,includes=*,append=true,output=file</jvm.args.jacoco>
```

Comment:
@zhfeng Could you remove coverage-reports directory and push files simply into ```${project.build.directory}``` throughout the pom or create that directory in test-resources phase somewhere? I think this is a relict of Narayana repo setup and it doesn't work if you don't have it:

I run it as (with CATALINA_HOME exported beforehand)
```
mvn clean integration-test -Parq-tomcat,codeCoverage
```

## Maven log
```
[INFO] --- maven-dependency-plugin:3.1.1:unpack-dependencies (unpack-jacoco-agent) @ narayana-tomcat ---
[INFO] Unpacking /home/karm/.m2/repository/org/jacoco/org.jacoco.agent/0.7.8/org.jacoco.agent-0.7.8.jar to /home/karm/mbabacek/workspaceRH/narayana-tomcat/tomcat-jta/target/lib with includes "" and excludes ""
```

## Tomcat log:
```
06-Sep-2018 12:27:48.025 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -javaagent:/home/karm/mbabacek/workspaceRH/narayana-tomcat/tomcat-jta/target/lib/jac     ocoagent.jar=destfile=/home/karm/mbabacek/workspaceRH/narayana-tomcat/tomcat-jta/target/coverage-reports/jacoco-ut.exec,includes=*,append=true,output=file
```
## No reports
```
karm@local:~/mbabacek/workspaceRH/narayana-tomcat (jbtm_jacoco)$ find -name "*.html"
./target/lib/about.html
./tomcat-jta/target/site/checkstyle.html
./tomcat-jta/target/lib/about.html
./tools/target/site/checkstyle.html
./tools/target/lib/about.html
```

Last but not least, little stupid minor thing, could it be formatted with newline and four spaces as
```
<jvm.args.jacoco>
    -javaagent....
...
```
?

Reply:
you need to run with
```
mvn clean install -Parq-tomcat,codeCoverage
```
since the maven-jacoco-plugin runs the report goal at the post-integration-test phase